jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -507,13 +507,9 @@ public boolean handle()
                                     status == HttpStatus.NO_CONTENT_204 ||
                                     status == HttpStatus.NOT_MODIFIED_304);
                                 if (hasContent && !_response.isContentComplete(_response.getHttpOutput().getWritten()))
-                                {
-                                    if (isCommitted())
-                                        abort(new IOException("insufficient content written"));
-                                    else
-                                        _response.sendError(HttpStatus.INTERNAL_SERVER_ERROR_500, "insufficient content written");
-                                }
+                                    sendErrorOrAbort("Insufficient content written");
                             }
+                            prepareUpgrade();
```

Comment:
There is no need to, since we don't have to setup anything on the read side _before_ we send a response back to the client.
With HTTP/2 we have to setup things before we send the response because we are always read interested, and the client can send tunnel data before we setup.